From 57c498fff66fd004e0ce2fe772e1de509d00c206 Mon Sep 17 00:00:00 2001
From: "Tan, Raymond" <raymond.tan@intel.com>
Date: Wed, 24 Jun 2015 15:18:22 +0800
Subject: [PATCH 333/389] gpio: gpio-pca953x: Handle register write for
 PCAL953X chip

This patch adds switch case to handle the PCAL953X chip type to
register write function.

Signed-off-by: Tan, Raymond <raymond.tan@intel.com>
Change-Id: Ib240298b349c99d0e4fa711aa51310b209b31191
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/5401
Reviewed-by: Fung, Joshua <joshua.fung@intel.com>
Reviewed-by: Ang, Chee Hong <chee.hong.ang@intel.com>
Tested-by: Ang, Chee Hong <chee.hong.ang@intel.com>
---
 drivers/gpio/gpio-pca953x.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpio/gpio-pca953x.c b/drivers/gpio/gpio-pca953x.c
index 9475dda..2f54a23 100644
--- a/drivers/gpio/gpio-pca953x.c
+++ b/drivers/gpio/gpio-pca953x.c
@@ -162,6 +162,7 @@ static int pca953x_write_regs(struct pca953x_chip *chip, int reg, u8 *val)
 			ret = i2c_smbus_write_word_data(chip->client,
 							reg << 1, (u16) *val);
 			break;
+		case PCAL953X_TYPE:
 		case PCA957X_TYPE:
 			ret = i2c_smbus_write_byte_data(chip->client, reg << 1,
 							val[0]);
-- 
1.9.3

